ui/src/index.css

```diff
@@ -16,6 +15,20 @@
   -webkit-text-size-adjust: 100%;
 }
 
+html, body {
+  /* ensure document root and body have no height, padding, or margin impact on react content */
+  height: 100%;
+  box-sizing: border-box;
+  padding: 0;
+  margin: 0;
+}
+
+#react-root {
+  height: 100%;
+  display: flex;
+  flex-direction: column;
```

Comment:
Assumes that the base layout will be a flex column. IMO it'd be fine to keep the react-root element styling unopininated and render an extra container layer in `layout.jsx` to keep the information about the layout's container close to the subcomponents making assumptions about it. Will leave that up to you.